packages/commonwealth/client/scripts/views/pages/contracts/add_contract_form.tsx

```diff
@@ -0,0 +1,224 @@
+/* @jsx m */
+
+import m from 'mithril';
+import $ from 'jquery';
+import Web3 from 'web3';
+
+import 'pages/create_community.scss';
+
+import app from 'state';
+import { MixpanelCommunityCreationEvent } from 'analytics/types';
+import { mixpanelBrowserTrack } from 'helpers/mixpanel_browser_util';
+import { initAppState } from 'app';
+import { slugifyPreserveDashes } from 'utils';
+import { ChainBase, ChainNetwork, ChainType, ContractType } from 'common-common/src/types';
+import { isAddress } from 'web3-utils';
+
+import { notifyError } from 'controllers/app/notifications';
+import { IdRow, InputRow, SelectRow } from 'views/components/metadata_rows';
+import CompoundAPI, {
+  GovernorTokenType,
+  GovernorType,
+} from 'controllers/chain/ethereum/compound/api';
+import AaveApi from 'controllers/chain/ethereum/aave/api';
+
+import { CWButton } from 'views/components/component_kit/cw_button';
+import { CWValidationText } from 'views/components/component_kit/cw_validation_text';
+
+import { linkExistingAddressToChainOrCommunity } from 'controllers/app/login';
+
+import {
+    initChainForm,
+    defaultChainRows,
+    ethChainRows,
+} from '../create_community/chain_input_rows';
+
+import {
+ChainFormFields,
+ChainFormState,
+EthChainAttrs,
+EthFormFields,
+} from '../create_community/types';
+
+type ContractFormFields = {
+  eth_chain_id: number,
+  abi: JSON,
+  contractType: ContractType.DAOFACTORY | ContractType.AAVE |ContractType.COMPOUND |
```

Comment:
remove DAOFACTORY here too. also, the form fields should probably just validate that it's a part of the enum (you shouldn't need to Union all the children of the enum here, there's better syntax)